### PR TITLE
Build iOS apps using Swift Packages

### LINKED ISSUE #57083: Crash when using Swift packages: xcodebuild: error: Packages are not supported when using legacy build locations, but the current project has them enabled
`flutter run`     

```bash
Launching lib/main.dart on iPhone 11 Pro Max in debug mode...
ProcessException: Process exited abnormally:
xcodebuild: error: Could not resolve package dependencies:
  Packages are not supported when using legacy build locations, but the current project has them
  enabled.
  Command: /usr/bin/xcodebuild -list
```

i have also try to change flutter channel  stable to beta

Doctor summary (to see all details, run flutter doctor -v):
```console
[✓] Flutter (Channel beta, v1.17.0, on Mac OS X 10.15.3 19D76, locale en-IN)
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 3.6)
[✓] VS Code (version 1.45.0)
[✓] Connected device (3 available)
```

### PR DESCRIPTION
**Pull request by jmagman**
## Description

Another attempt at https://github.com/flutter/flutter/pull/58522.

Swift packages cannot be added to a newly created Flutter iOS app due to the Xcode project settings (not workspace) being set to the "legacy" build location.  Update to the "default" Xcode build location.
73312223-10950180-41dd-11ea-9a66-98d3a140f0fd

Screen Shot 2020-06-02 at 1 56 59 PM

Update templates, add tool migrator, let tool upgrade all examples and integration tests.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/57083
Related to https://github.com/flutter/flutter/pull/49654

## Tests
- Migrator tests

- Added the [basic Apple example Swift package](https://github.com/apple/example-package-playingcard) to an existing integration test.  The adds a networked dependency to integration test, but hopefully Apple never removes this (and if they do, we can replace with another one).

## Checklist
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change
- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*

### CODE REVIEW

**Comment by jmagman** [comment]
The difference between this and #58522 is https://github.com/flutter/flutter/pull/59009/commits/cb5b9e2d3da0aa7738313535ae625225172d72dc
- Change `"group:Runner.xcodeproj"` to `location = "self:"` (#58522 had `self:Runner.xcodeproj`)
- Remove `location = "self:Pods/Pods.xcodeproj"` (not sure why the _project_ ever needed to know about it since it's in the workspace).
- Support migrating lines and entire files sequentially.

#58522 `ios_app_with_extensions_test` failed on devicelab post-submit, though it succeeded on my machine.  That test on this PR also succeeds on my machine, so we'll have to see what devicelab thinks.

File: packages/flutter_tools/lib/src/ios/migrations/project_build_location_migration.dart [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  String migrateLine(String line) {
+    const String legacyBuildLocation = 'location = "group:Runner.xcodeproj"';
+    const String defaultBuildLocation = 'location = "self:"';
+
+    return line.replaceAll(legacyBuildLocation, defaultBuildLocation);
+  }
+
+  @override
+  String migrateFileContents(String fileContents) {
+    const String podLocation = '''
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/ios/migrations/project_build_location_migration.dart:41]: nit. is this guaranteed to match, even with the surrounding whitespace?

**Review by christopherfujino — APPROVED** [review]
LGTM, with one nit question

File: packages/flutter_tools/lib/src/ios/migrations/project_build_location_migration.dart (hunk 1 above)
Comment (jmagman) [reply packages/flutter_tools/lib/src/ios/migrations/project_build_location_migration.dart:41]: We generated this as part of the template with https://github.com/flutter/flutter/pull/4893/files#diff-fddbdff9cdb9e32ee9b9a95f74f30788R5-R8.  Xcode and CocoaPods won't add it (at least how we have the project set up now).

**Pull request merged**